src/test/scala/chiselTests/AsyncResetSpec.scala

```diff
@@ -0,0 +1,104 @@
+// See LICENSE for license details.
+
+package chiselTests
+
+import chisel3._
+import chisel3.experimental.{AsyncReset, withClockAndReset, withClock, withReset, Reset}
+import chisel3.util.{Counter, Queue}
+import chisel3.testers.BasicTester
+
+class AsyncResetTester extends BasicTester {
+  val (_, cDiv) = Counter(true.B, 4)
+  // First rising edge when count === 3
+  val slowClk = cDiv.asClock
+
+  val (count, done) = Counter(true.B, 8)
+
+  val asyncResetNext = RegInit(false.B)
+  asyncResetNext := count === 0.U
+  val asyncReset = asyncResetNext.asAsyncReset
+
+  val reg = withClockAndReset(slowClk, asyncReset) {
+    RegInit(123.U(8.W))
+  }
+  reg := 5.U // Normal connection
+
+  when (count > 0.U && count < 3.U) {
+    assert(reg === 123.U)
+  } .elsewhen (count >= 3.U) {
+    assert(reg === 5.U)
+  }
+
+  when (done) {
+    stop()
+  }
+}
+
+class AsyncResetQueueTester extends BasicTester {
```

Comment:
Is this a common use for a Queue? Because it's not a clock crossing queue, right, it's just a async-reset queue?

Reply:
Queues aren't special here, this is just illustrating that for modules written the right way, you can use them as either synchronous or asynchronous reset just by the context within which they are used. This is in contrast to Verilog where you have to rewrite all of the always blocks to accomplish the same goal.